tests/test_machine.py

```diff
@@ -0,0 +1,53 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import os
+from subprocess import TimeoutExpired
+
+import pytest
+
+import mkosi.machine as machine
+from mkosi.backend import MkosiException
+
+
+pytestmark = [pytest.mark.integration, pytest.mark.parametrize("verb", ["boot", "qemu"]), pytest.mark.skipif(os.getuid() != 0, reason="Must be invoked as root.")]
+
+
+def test_simple_run(verb: str) -> None:
+    with machine.Machine([verb]) as m:
+        p = m.run(["echo", "This is a test."])
+        assert "This is a test." == p.stdout.strip("\n")
+
+    assert m.exit_code == 0
+
+
+def test_wrong_command(verb: str) -> None:
+    with machine.Machine([verb]) as m:
+        with pytest.raises(MkosiException):
+            m.run(["NonExisting", "Command"], check=True)
+        with pytest.raises(MkosiException):
+            m.run(["ls", "NullDirectory"], check=True)
```

Comment:
We should also test with `check=False`. And `check=True` is the default so no need to repeat it here.